### DISCUSSION TITLE
SimpleClient - expand usage beyond ASType

### DISCUSSION THREAD

**Discussion by CreggHancock** [warriordog/ActivityPubSharp#112] [Ideas]
This is more of a rough idea for a feature proposal for now before I create an actual issue for it, but would it be a good idea to allow for deserializing different types than the ASType type?

As background, I went to test the SimpleClient for the first time and was "open"-ing my  "Person" type from my mastodon profile. I then tried printing the various properties in my profile and thought I ran into a bug when many of them were throwing errors when I attempted to print them. Then after digging into the code I saw we are deserializing into the base ASType, so only a couple of the properties were actually being parsed into the object. 

Is it possible we can allow fetching of various types so we can test out viewing different Actors and Objects via command line parameters? I can think of a couple of ways to go about this, but they would all be pretty verbose so I'm totally open to ideas.

If this was already the plan or if there was something wrong with how I'm using the client and I should already be able to access those properties then please let me know, and maybe I can add some of that knowledge to the "help" command in the future.

**Comment by warriordog** [comment]
Great suggestion! That's actually how SimpleClient used to work, before we implemented the TypeMap system. I wonder if a more user-friendly approach would be to deserialize to TypeMap and then introspect all of the loaded entities together? Then its possible to access properties from *any* type in the graph.

**Comment by warriordog** [comment]
I've created #115 to address this.
